Flaky integration tests of verifying the replica number of HA

It did not happen every time, but this flaky test has been in for a long time.
We can solve it by adding a retry mechanism.
```
=== RUN   TestKnativeServingDeployment/configure
    verify.go:297: KnativeServing "knative" failed to update: Operation cannot be fulfilled on knativeservings.operator.knative.dev "knative": the object has been modified; please apply your changes to the latest version and try again
--- FAIL: TestKnativeServingDeployment/configure (40.60s)
FAIL test/e2e.TestKnativeServingDeployment/configure (40.60s)
PASS test/e2e.TestKnativeServingDeployment/restore (10.34s)
PASS test/e2e.TestKnativeServingDeployment/delete (20.45s)
=== RUN   TestKnativeServingDeployment
--- FAIL: TestKnativeServingDeployment (111.65s)
FAIL test/e2e.TestKnativeServingDeployment (111.65s)
FAIL test/e2e
=== Failed
=== FAIL: test/e2e TestKnativeServingDeployment/configure (40.60s)
    verify.go:297: KnativeServing "knative" failed to update: Operation cannot be fulfilled on knativeservings.operator.knative.dev "knative": the object has been modified; please apply your changes to the latest version and try again
=== FAIL: test/e2e TestKnativeServingDeployment (111.65s)
```

--- a/test/resources/verify.go
+++ b/test/resources/verify.go
@@ -286,15 +286,24 @@ func AssertKEOperatorCRReadyStatus(t *testing.T, clients *test.Clients, names te
 
 // VerifyHADeployments verify whether all the deployments has scaled up.
 func VerifyHADeployments(t *testing.T, clients *test.Clients, names test.ResourceNames) {
-	ks, err := clients.KnativeServing().Get(context.TODO(), names.KnativeServing, metav1.GetOptions{})
+	err := wait.PollImmediate(Interval, Timeout, func() (bool, error) {
+		ks, err := clients.KnativeServing().Get(context.TODO(), names.KnativeServing, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("KnativeServing %q failed to get: %v", names.KnativeServing, err)
+		}
+		var two int32 = 2
+		ks.Spec.HighAvailability = &base.HighAvailability{Replicas: &two}
+		_, err = clients.KnativeServing().Update(context.TODO(), ks, metav1.UpdateOptions{})
+		if err != nil {
+			t.Logf("KnativeServing %q failed to update: %v", names.KnativeServing, err)
+			t.Logf("Retrying...")
+			return false, nil
+		}
+		return true, nil
+	})
+
 	if err != nil {
-		t.Fatalf("KnativeServing %q failed to get: %v", names.KnativeServing, err)
-	}
-	var two int32 = 2
-	ks.Spec.HighAvailability = &base.HighAvailability{Replicas: &two}
-	_, err = clients.KnativeServing().Update(context.TODO(), ks, metav1.UpdateOptions{})
-	if err != nil {
-		t.Fatalf("KnativeServing %q failed to update: %v", names.KnativeServing, err)
+		t.Fatal("Timed out updating the HA on KnativeServing", err)
 	}
 
 	err = wait.PollImmediate(Interval, Timeout, func() (bool, error) {
@@ -315,11 +324,11 @@ func VerifyHADeployments(t *testing.T, clients *test.Clients, names test.Resourc
 		return true, nil
 	})
 	if err != nil {
-		t.Fatal("Timed out waiting on KnativeServing to delete", err)
+		t.Fatal("Timed out waiting on KnativeServing to update the deployments", err)
 	}
 
 	// Get KnativeServing CR again to avoid "the object has been modified" error.
-	ks, err = clients.KnativeServing().Get(context.TODO(), names.KnativeServing, metav1.GetOptions{})
+	ks, err := clients.KnativeServing().Get(context.TODO(), names.KnativeServing, metav1.GetOptions{})
 	if err != nil {
 		t.Fatalf("KnativeServing %q failed to get: %v", names.KnativeServing, err)
 	}

